src/ImageCapture_Gst.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * This file is part of the Camera Streaming Daemon
+ *
+ * Copyright (C) 2017  Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+#include <string>
+
+#include "ImageCapture.h"
+
+class ImageCapture_Gst final : public ImageCapture {
+public:
+    ImageCapture_Gst(std::shared_ptr<CameraDevice> camDev);
+    ~ImageCapture_Gst();
+    int start(int interval, int count, std::function<void(int result, int seq_num)> cb);
+    int stop();
+    int getState();
+    int setResolution(int imgWidth, int imgHeight);
+    int setFormat(int imgFormat);
+    int setDestination(const std::string imgPath);
+
+private:
+    int setState(int state);
+    int click(int seq_num);
+    void capture_thread(int num, int interval);
+    int create_v4l2_pipeline(int seq_num);
+    std::string getGstElementName_format(int format);
+    std::string getExtension(int format);
+    std::string getGstPipelineName_v4l2(int seq_num);
+    std::shared_ptr<CameraDevice> mCamDev;
+    std::string mDevice;
+    std::atomic<int> mState;
+    int mWidth;
+    int mHeight;
+    int mImgFormat;
```

Comment:
Its better to use `enum` for image format & pixel format over `int`, which increases readability.

Reply:
Possible values for the formats are defined as const integers, hence int